discovery/aws/ecs_test.go

```diff
@@ -0,0 +1,758 @@
+						},
+					},
+				},
+			},
+			clusterARNs: []string{"arn:aws:ecs:us-west-2:123456789012:cluster/test-cluster"},
+			expected: map[string]ecsTypes.Cluster{
+				"arn:aws:ecs:us-west-2:123456789012:cluster/test-cluster": {
+					ClusterName: strptr("test-cluster"),
+					ClusterArn:  strptr("arn:aws:ecs:us-west-2:123456789012:cluster/test-cluster"),
+					Status:      strptr("ACTIVE"),
+					Tags: []ecsTypes.Tag{
+						{Key: strptr("Environment"), Value: strptr("test")},
+						{Key: strptr("Team"), Value: strptr("backend")},
+					},
+				},
+			},
+		},
+		{
+			name: "MultipleClusters",
+			ecsData: &ecsDataStore{
+				region: "us-east-1",
+				clusters: []ecsTypes.Cluster{
+					{
+						ClusterName: strptr("cluster-1"),
+						ClusterArn:  strptr("arn:aws:ecs:us-east-1:123456789012:cluster/cluster-1"),
+						Status:      strptr("ACTIVE"),
+					},
+					{
+						ClusterName: strptr("cluster-2"),
+						ClusterArn:  strptr("arn:aws:ecs:us-east-1:123456789012:cluster/cluster-2"),
+						Status:      strptr("DRAINING"),
+						Tags: []ecsTypes.Tag{
+							{Key: strptr("Stage"), Value: strptr("prod")},
+						},
+					},
+				},
+			},
+			clusterARNs: []string{
+				"arn:aws:ecs:us-east-1:123456789012:cluster/cluster-1",
+				"arn:aws:ecs:us-east-1:123456789012:cluster/cluster-2",
+			},
+			expected: map[string]ecsTypes.Cluster{
+				"arn:aws:ecs:us-east-1:123456789012:cluster/cluster-1": {
+					ClusterName: strptr("cluster-1"),
+					ClusterArn:  strptr("arn:aws:ecs:us-east-1:123456789012:cluster/cluster-1"),
+					Status:      strptr("ACTIVE"),
+				},
+				"arn:aws:ecs:us-east-1:123456789012:cluster/cluster-2": {
+					ClusterName: strptr("cluster-2"),
+					ClusterArn:  strptr("arn:aws:ecs:us-east-1:123456789012:cluster/cluster-2"),
+					Status:      strptr("DRAINING"),
+					Tags: []ecsTypes.Tag{
+						{Key: strptr("Stage"), Value: strptr("prod")},
+					},
+				},
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			client := newMockECSClient(tt.ecsData)
+
+			d := &ECSDiscovery{
+				ecs: client,
+				cfg: &ECSSDConfig{
+					Region:             tt.ecsData.region,
+					RequestConcurrency: 10,
+				},
+			}
+
+			clusterMap, err := d.describeClusters(ctx, tt.clusterARNs)
+			require.NoError(t, err)
+			require.Equal(t, tt.expected, clusterMap)
+		})
+	}
+}
+
+func TestECSDiscoveryListServiceARNs(t *testing.T) {
+	ctx := context.Background()
+
+	// iterate through the test cases
+	for _, tt := range []struct {
+		name        string
+		ecsData     *ecsDataStore
+		clusterARNs []string
+		expected    map[string][]string
+	}{
+		{
+			name: "SingleClusterWithServices",
+			ecsData: &ecsDataStore{
+				region: "us-west-2",
+				clusters: []ecsTypes.Cluster{
+					{
+						ClusterName: strptr("test-cluster"),
+						ClusterArn:  strptr("arn:aws:ecs:us-west-2:123456789012:cluster/test-cluster"),
+						Status:      strptr("ACTIVE"),
+					},
+				},
+				services: []ecsTypes.Service{
```

Comment:
You might want to add an exampe of an old-style ARN where the ARN doesn't contain the cluster name and tags are not supported for completeness: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-arn-migration.html